packages/mdc-segmented-button/segmented-button/component.ts

```diff
@@ -21,45 +21,113 @@
  * THE SOFTWARE.
  */
 
-import {MDCComponent} from '@material/base/component';
-import {MDCSegmentedButtonSegmentFactory} from '../segment/component';
-import {SegmentDetail} from '../types';
+import {MDCComponent} from '../../mdc-base/component';
+import {CustomEventListener} from '../../mdc-base/types';
+import {MDCSegmentedButtonSegment, MDCSegmentedButtonSegmentFactory} from '../segment/component';
+import {SegmentDetail, MDCSegmentedButtonEvent} from '../types';
+import {MDCSegmentedButtonAdapter} from './adapter';
 import {MDCSegmentedButtonFoundation} from './foundation';
+import {strings} from './constants';
 
 export class MDCSegmentedButton extends MDCComponent<MDCSegmentedButtonFoundation> {
   static attachTo(root: Element): MDCSegmentedButton {
     return new MDCSegmentedButton(root);
   }
 
-  initialize(_segmentFactory: MDCSegmentedButtonSegmentFactory) {
-    return;
+  private segments!: MDCSegmentedButtonSegment[]; // assigned in initialize
+  private segmentFactory!:
+      (el: Element) => MDCSegmentedButtonSegment; // assigned in initialize
+  private handleSelected!:
+      CustomEventListener<MDCSegmentedButtonEvent>; // assigned in initialSyncWithDOM
+
+  initialize(segmentFactory: MDCSegmentedButtonSegmentFactory = (el) => new MDCSegmentedButtonSegment(el)) {
+    this.segmentFactory = segmentFactory;
+    this.segments = this.instantiateSegments(this.segmentFactory);
+  }
+
+  private instantiateSegments(segmentFactory: MDCSegmentedButtonSegmentFactory): MDCSegmentedButtonSegment[] {
+    const segmentElements: Element[] =
+        [].slice.call(this.root.querySelectorAll(strings.SEGMENT_SELECTOR));
+    return segmentElements.map((el: Element, index: number) => {
+      let segment = segmentFactory(el);
+      segment.setIndex(index);
+      segment.setIsSingleSelect(this.foundation.isSingleSelect());
+      return segment;
+    });
   }
 
   initialSyncWithDOM() {
-    return;
+    this.handleSelected = (event) =>
+        this.foundation.handleSelected(event.detail);
+    this.listen(strings.SELECTED_EVENT, this.handleSelected);
   }
 
   destroy() {
+    this.listen(strings.SELECTED_EVENT, this.handleSelected);
+
     super.destroy();
   }
 
   getDefaultFoundation(): MDCSegmentedButtonFoundation {
-    return new MDCSegmentedButtonFoundation();
+    const adapter: MDCSegmentedButtonAdapter = {
+      hasClass: (className: string): boolean => {
```

Comment:
Do you need to specify the explicit return type?

Reply:
It's not necessary, I do it to avoid type errors. It looks like other components don't do this in `getDefaultFoundation`, so I'll remove them.

Reply:
TS is pretty good at type inference so you generally don't need declare explicit return types like this. For more complex functions, it could be useful for more complex functions.

Reply:
I'd defer to Ahbinay on this - some of the external tooling (eg for documentation extraction) does not work well without explicit return types.

Reply:
TS does not mandate to specify explicit return type in this case. It infers type information from `MDCSegmentedButtonAdapter`.